assets/javascripts/discourse/components/global-filter/composer-container.js

```diff
@@ -17,4 +19,23 @@ export default class GlobalFilterComposerContainer extends Component {
     }
     return filters.split("|");
   }
+
+  constructor() {
+    super(...arguments);
+
+    let categories = [];
+    withPluginApi("1.3.0", (api) => {
+      ajax(
+        `/global_filter/filter_tags/categories_for_current_filter.json`
+      ).then((model) => {
+        categories = model.categories;
+      });
+
+      api.modifySelectKit("category-chooser").replaceContent((component) => {
```

Comment:
Would it make sense to call `replaceContent` inside the Ajax call? Otherwise, you could be replacing the category list with an empty array of categories, sometimes, if the ajax call is too slow.